packages/strapi-admin/services/permission/sections-builder/handlers.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const { curry, matchesProperty, pick } = require('lodash/fp');
+
+const utils = {
+  isOfKind: kind => matchesProperty('kind', kind),
+  resolveContentType: uid => strapi.contentTypes[uid],
+  isNotInSubjects: subjects => uid => !subjects.find(subject => subject.uid === uid),
+  hasProperty: curry((property, subject) => {
+    return !!subject.properties.find(prop => prop.value === property);
+  }),
+  getValidOptions: pick(['applyToProperties']),
+  toSubjectTemplate: ct => ({ uid: ct.uid, label: ct.info.name, properties: [] }),
+};
+
+/**
+ * Transforms & adds the given  setting action to the section
+ * Note: The action is transformed to a setting specific format
+ */
+const settings = ({ action, section }) => {
+  const { category, subCategory, displayName, actionId } = action;
+
+  section.push({
+    displayName,
+    category,
+    subCategory,
+    action: actionId,
+  });
+};
+
+/**
+ * Transforms & adds the given plugin action to the section
+ * Note: The action is transformed to a plugin specific format
+ */
+const plugins = ({ action, section }) => {
+  const { pluginName, subCategory, displayName, actionId } = action;
+
+  section.push({
```

Comment:
section.push here & just under section.actions.push ? are these the same section objects ?

Reply:
They're not, they don't have the same shape.

Ref: https://github.com/strapi/strapi/pull/9535#discussion_r592251820

Reply:
change the naming / jsdoc so we know one section is an array of action & the other is an object